local-volume/bootstrapper/bootstrapper.go

```diff
@@ -67,13 +71,48 @@ func setupClient() *kubernetes.Clientset {
 	return clientset
 }
 
-func generateMountName(hostDir, mountDir string) string {
+func generateMountName(mount *common.MountConfig) string {
 	h := fnv.New32a()
-	h.Write([]byte(hostDir))
-	h.Write([]byte(mountDir))
+	h.Write([]byte(mount.HostDir))
+	h.Write([]byte(mount.MountDir))
 	return fmt.Sprintf("mount-%x", h.Sum32())
 }
 
+// generateMountDir generates mount directory path in container, rule is to trim '/' prefix and
+// change "/" to "~", according to kubernetes convention, then concatenate with default root path.
+// E.g.
+//   "/mnt/ssds" -> "/mnt/local-storage/mnt~ssds"
+func generateMountDir(mount *common.MountConfig) string {
+	return path.Join(*mountRoot, strings.Replace(strings.TrimPrefix(mount.HostDir, "/"), "/", "~", -1))
+}
+
+func ensureMountDir(config map[string]common.MountConfig) bool {
+	needsUpdate := false
+	for class, mount := range config {
+		if mount.MountDir == "" {
+			newMoutConfig := mount
+			newMoutConfig.MountDir = generateMountDir(&mount)
+			config[class] = newMoutConfig
+			needsUpdate = true
+		}
+	}
+	return needsUpdate
+}
+
+func updateConfigMap(client *kubernetes.Clientset, name, namespace string, config map[string]common.MountConfig) error {
+	var err error
+	configMap, err := client.CoreV1().ConfigMaps(namespace).Get(name, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+	configMap.Data, err = common.VolumeConfigToConfigMapData(config)
+	if err != nil {
+		return err
+	}
+	_, err = client.CoreV1().ConfigMaps(namespace).Update(configMap)
```

Comment:
I'm slightly weary of updating a user-created config map.  Maybe it would be better to generate a new configmap?